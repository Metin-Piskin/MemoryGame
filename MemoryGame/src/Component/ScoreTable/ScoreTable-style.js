import { StyleSheet } from "react-native";

export default StyleSheet.create({
    innercontainer: {
        flexDirection: 'row',
        justifyContent: 'space-around'
    },
    textName: {
        paddingHorizontal: 40,
        borderWidth: 1,
        fontFamily: 'PressStart2P-Regular',
        textAlign: 'center',
        textAlignVertical: 'bottom',
        paddingVertical: 10,
        color: '#fff',
        borderColor: '#fff'
    },
    textDifficulty: {
        paddingHorizontal: 18,
        borderWidth: 1,
        fontFamily: 'PressStart2P-Regular',
        textAlign: 'center',
        textAlignVertical: 'bottom',
        paddingVertical: 10,
        color: '#fff',
        borderColor: '#fff'
    },
    textScore: {
        paddingHorizontal: 15,
        borderWidth: 1,
        fontFamily: 'PressStart2P-Regular',
        textAlign: 'center',
        textAlignVertical: 'bottom',
        paddingVertical: 10,
        color: '#fff',
        borderColor: '#fff'
    },
})