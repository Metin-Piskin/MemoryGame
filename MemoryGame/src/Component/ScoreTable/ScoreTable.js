import React from 'react';
import { View, Text } from 'react-native';

import styles from './ScoreTable-style';

const ScoreTable = () => {
    return (
        <View style={styles.innercontainer} testID='score-table-innercontainer'>
            <Text style={styles.textName} testID='score-table-textName'>Name</Text>
            <Text style={styles.textDifficulty} testID='score-table-textDifficulty'>Difficulty</Text>
            <Text style={styles.textScore} testID='score-table-textScore'>Scor</Text>
        </View>
    )
}

export default ScoreTable;