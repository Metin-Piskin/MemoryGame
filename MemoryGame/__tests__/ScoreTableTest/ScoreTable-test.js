import React from 'react';
import { render } from '@testing-library/react-native';

import ScoreTable from '../../src/Component/ScoreTable';
import ScoreTableStyles from '../../src/Component/ScoreTable/ScoreTable-style';


test('renders correctly', () => {
    const comp = render(<ScoreTable />);
    expect(comp).toMatchSnapshot();
});

test('should render default theme style', () => {
    const comp = render(<ScoreTable />);
    const ScoreTableStyle = comp.getByTestId('score-table-innercontainer').props.style;

    expect(ScoreTableStyle).toMatchObject(ScoreTableStyles.innercontainer);
});

test('should render default theme style', () => {
    const comp = render(<ScoreTable />);
    const ScoreTableStyle = comp.getByTestId('score-table-textName').props.style;

    expect(ScoreTableStyle).toMatchObject(ScoreTableStyles.textName);
});

test('should render default theme style', () => {
    const comp = render(<ScoreTable />);
    const ScoreTableStyle = comp.getByTestId('score-table-textDifficulty').props.style;

    expect(ScoreTableStyle).toMatchObject(ScoreTableStyles.textDifficulty);
});

test('should render default theme style', () => {
    const comp = render(<ScoreTable />);
    const ScoreTableStyle = comp.getByTestId('score-table-textScore').props.style;

    expect(ScoreTableStyle).toMatchObject(ScoreTableStyles.textScore);
});
