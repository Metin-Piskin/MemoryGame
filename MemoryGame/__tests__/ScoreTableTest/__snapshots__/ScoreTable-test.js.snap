// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "justifyContent": "space-around",
    }
  }
  testID="score-table-innercontainer"
>
  <Text
    style={
      {
        "borderColor": "#fff",
        "borderWidth": 1,
        "color": "#fff",
        "fontFamily": "PressStart2P-Regular",
        "paddingHorizontal": 40,
        "paddingVertical": 10,
        "textAlign": "center",
        "textAlignVertical": "bottom",
      }
    }
    testID="score-table-textName"
  >
    Name
  </Text>
  <Text
    style={
      {
        "borderColor": "#fff",
        "borderWidth": 1,
        "color": "#fff",
        "fontFamily": "PressStart2P-Regular",
        "paddingHorizontal": 18,
        "paddingVertical": 10,
        "textAlign": "center",
        "textAlignVertical": "bottom",
      }
    }
    testID="score-table-textDifficulty"
  >
    Difficulty
  </Text>
  <Text
    style={
      {
        "borderColor": "#fff",
        "borderWidth": 1,
        "color": "#fff",
        "fontFamily": "PressStart2P-Regular",
        "paddingHorizontal": 15,
        "paddingVertical": 10,
        "textAlign": "center",
        "textAlignVertical": "bottom",
      }
    }
    testID="score-table-textScore"
  >
    Scor
  </Text>
</View>
`;
